### PR TITLE
fix stop mojo by taking container name pattern into account

### PR DESCRIPTION
**Pull request by j3t**
Fixes fabric8io/docker-maven-plugin#1168

### CODE REVIEW

**Comment by zbelzer** [comment]
This would be very useful for our team. Please consider its review.

**Review by ohbus — APPROVED** [review]

File: src/main/java/io/fabric8/maven/docker/util/ContainerNamingUtil.java [hunk 1]
```diff
@@ -57,33 +57,37 @@ public static String formatContainerName(final ImageConfiguration image,
...
-                image.getAlias(),
-                buildTimestamp);
+        Map<String, FormatParameterReplacer.Lookup> lookups = new HashMap<>();
+        lookups.put("a", () -> image.getAlias());
+        lookups.put("n", () -> cleanImageName(image.getName()));
+        lookups.put("t", () -> "\\d{10,}");
+        lookups.put("i", () -> "\\d+");
 
-        return keepOnlyLastIndexedContainer(containers, partiallyApplied);
+        String appliedContainerNamePattern = new FormatParameterReplacer(lookups).replace(containerNamePattern);
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/util/ContainerNamingUtil.java:86]: Is it correct that this changes the current behaviour in a non-backwards compatible way? (i.e. that all containers are stopped now by default, not the latest started)

If so we would need to be more careful to keep the old behaviour (at least for a certain deprecation period) and introduce this new behaviour either by a certain indication in the container name pattern or if this is not possible, with a new flag for `docker:stop`.

**Review by rhuss — COMMENTED** [review]
In general I believe the use case that this PR is solving and that is described in the reference issue is valid, but I'm concerned about backwards compatibility which must be ensured.

Unfortunately I don't have enough time to dive deeper in this PR technically, but @rohanKanojia if you could give it tech review that would be cool.

**Comment by rhuss** [comment]
btw, I'm also not sure why only the container name with the highest index is used (when the container pattern includes a `%i`. We might remove it, though.

**Comment by rhuss** [comment]
Can we get #1410 in first and then check out how this fit to this PR ? (guess there might be a conflict)

**Comment by rohanKanojia** [comment]
@j3t: Hello, Thanks for updating this PR. Somehow commit history seems to be messed up. Could you please take a look into it? Let's try to get it reviewed and merged soon.

**Comment by j3t** [comment]
Hi @rohanKanojia, is there any chance this gets included in the next release? What is missing? How can I help?

**Comment by rhuss** [comment]
Looks good to me.  Let's get that merged.

**Comment by rhuss** [comment]
@j3t thanks a ton ! Sorry for the delay, I hope we can make a release asap.

**Comment by rohanKanojia** [comment]
@j3t: I'm sorry for the late reply. I was on PTO. Do you need a new patch release with this fix? Or are you okay with waiting for the next minor release?

**Comment by j3t** [comment]
> @j3t: I'm sorry for the late reply. I was on PTO. Do you need a new patch release with this fix? Or are you okay with waiting for the next minor release?

No worries, the next minor release is sufficient I guess.

**Pull request merged**